ibis/pandas/core.py

```diff
@@ -253,16 +298,19 @@ def execute_until_in_scope(
     # base case: our op has been computed (or is a leaf data node), so
     # return the corresponding value
     op = expr.op()
-    if op in scope:
-        return scope
-    elif isinstance(op, ops.Literal):
+    if isinstance(op, ops.Literal):
         # special case literals to avoid the overhead of dispatching
         # execute_node
-        return {
-            op: execute_literal(
+        return set_scope_item(
+            op,
+            execute_literal(
                 op, op.value, expr.type(), aggcontext=aggcontext, **kwargs
-            )
-        }
+            ),
+            timecontext,
+        )
+
+    if get_scope_item(scope, op, timecontext) is not None:
```

Comment:
Why change the order to be after Literal check here?

Reply:
Good catch, this is a bug.